### PR TITLE
Backup and restore for v2 CloudSQL and GKE

### PR DESCRIPTION
**Pull request by filiphaftek**
[issue](https://github.com/sourcegraph/customer/issues/1356)
Backup and restore for:
- CloudSQL
- GKE cluster
- single application (asssuming disk (PVC) was not deleted)

For single application with deleted PVC I will make separate PR with [customer logic](https://cloud.google.com/kubernetes-engine/docs/add-on/backup-for-gke/how-to/protected-application#backup-one-restore-all)

### CODE REVIEW

File: content/departments/cloud/technical-docs/v2.0/restore_process.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+Use cases:
+
+- Cloud SQL data is corrupted by a broken database migration
+- Cloud SQL data is deleted
+
+### Restore from automated backup
+
+> Below process is derived from [GCP documentation](https://cloud.google.com/sql/docs/postgres/backup-recovery/restoring#gcloud)
+
+The restortion process will be performed with `gcloud`. Learn more about [why not terraform?](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#restore_backup_context).
```
Comment (rafax) [inline content/departments/cloud/technical-docs/v2.0/restore_process.md]:
```suggestion
The restoration process will be performed with `gcloud`. Learn more about [why not terraform?](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#restore_backup_context).
```

File: content/departments/cloud/technical-docs/v2.0/restore_process.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+> NOTE: **IMPORTANT** Backup has to be enabled during [creation process](./creation_process.md#enable-backup)
+
+1. List backups
+
+```sh
+mi2 instance backup list --slug <CUSTOMER> -e [dev|prod]
+```
+
+2. Create on-demand backup
```
Comment (rafax) [inline content/departments/cloud/technical-docs/v2.0/restore_process.md]: Why are we creating a backup during restore - is this supposed to be used when restore goes wrong?

Comment (filiphaftek) [reply content/departments/cloud/technical-docs/v2.0/restore_process.md]: No, I just used it during my tests to have fresh backup. I remove it from here, b/c we have daily backups same as for CloudSQL, so on-demand can go to FAQ.

**Review by michaellzc — APPROVED** [review]
lgtm

**Review by michaellzc — COMMENTED** [review]
nit: replace <CUSTOMER> <ENVIRONMENT> with `$CUSTOMER` and `$ENVIRONMENT`

- consistent with references in creation process
- most command we can just copy and paste

**Comment by filiphaftek** [comment]
> $ENVIRONMENT

I replaced with $ENVIRONMENT and $SLUG, b/c you used it also in creation process. I do not think we should use CUSTOMER, as we have parameter called --slug :)

**Pull request merged**